### PR TITLE
REFACTOR: Remove defaults to pandas at API layer and add some corresponding client QC methods.

### PR DESCRIPTION
**Pull request by mvashishtha**
Signed-off-by: mvashishtha <mahesh@ponder.io>

## What do these changes do?

- [ ] passes `flake8 modin`
- [ ] passes `black --check modin`
- [ ] signed commit with `git commit -s` 
- [ ] Resolves #? 
- [ ] tests added and passing

### CODE REVIEW

**Comment by mvashishtha** [comment]
I'm testing the changes in https://github.com/ponder-org/repo-name/pull/104

File: modin/_compat/pandas_api/latest/base.py [hunk 1]
```diff
@@ -241,7 +241,7 @@ def rolling(
             closed=closed,
             method=method,
         )
-
+
```
Comment (naren-ponder) [inline modin/_compat/pandas_api/latest/base.py]: Nit: Can we remove this change unless it is needed by the linter?

File: modin/pandas/dataframe.py [hunk 2]
```diff
@@ -2555,7 +2528,7 @@ def __round__(self, decimals=0):
         -------
         DataFrame
         """
-        return self._default_to_pandas(pandas.DataFrame.__round__, decimals=decimals)
+        return self.round(decimals)
```
Comment (naren-ponder) [inline modin/pandas/dataframe.py:2531]: Does this need `self._query_compiler.round`?

**Review by naren-ponder — COMMENTED** [review]
LGTM overall, just some minor comments.

File: modin/pandas/dataframe.py (hunk 2 above)
Comment (mvashishtha) [reply modin/pandas/dataframe.py:2531]: I think API level round round should work.

File: modin/_compat/pandas_api/latest/base.py (hunk 1 above)
Comment (mvashishtha) [reply modin/_compat/pandas_api/latest/base.py]: Done.

**Comment by mvashishtha** [comment]
@naren-ponder ready for review again

**Review by naren-ponder — APPROVED** [review]

**Pull request merged**